### PR TITLE
Fix #15609

### LINKED ISSUE #15609: "imagePullPolicy: Always" does not re-pull an image
Consider the following Dockerfile:

```
FROM debian:jessie
RUN echo 1 > /counter
CMD ["bash", "-c", "for ((i = 0; ; i++)); do cat /counter; sleep 1; done"]
```

And the following rc.yaml:

```
apiVersion: v1
kind: ReplicationController
metadata:
  name: ipp-test
spec:
  replicas: 2
  selector:
    app: ipp-test
    deployment: initial
  template:
    metadata:
      labels:
        app: ipp-test
        deployment: initial
    spec:
      containers:
      - name: ipp-test
        image: us.gcr.io/project-108021/ipp_test
      imagePullPolicy: Always
```

Now I say

```
docker build -t us.gcr.io/project-108021/ipp_test .
gcloud docker --server=us.gcr.io push us.gcr.io/project-108021/ipp_test
kubectl create -f rc.yaml
```

After changing "echo 1" to "echo 2" in the Dockerfile and re-building and re-pushing it to the same image name/tag, I call

```
kubectl rolling-update ipp-test --image=us.gcr.io/project-108021/ipp_test
```
and watch the logs with
```
kubectl logs ipp-test-<yourPodHashHere>
```

... and get a lot of "1"s.  Thus, the image was not pulled although imagePullPolicy was set to "Always".  I see no difference between "Always" and "IfNotPresent" then, so I suspect something in kubectl is not working.

### PR DESCRIPTION
**Pull request by bgrant0607**
cc @caesarxuchao @nikhiljindal

### CODE REVIEW

**Comment by erictune** [comment]
LGTM

File: docs/user-guide/working-with-resources.md [hunk 1]
```diff
@@ -75,23 +75,13 @@ The system adds fields in several ways:
...
-  - Go to `http://localhost:8001/swagger-ui` in your browser.
-  - It should say "swagger" at the top-left.
-
-Once there:
-  - Click on "v1" and wait for it to expand.
-  - Search for "pods", "services", "replicationcontrollers" or some other resource.
-  - Click on that POST row for the matching resource.
-  - Click on the words "Model".
-  - You should see a list of all possible resource fields, starting with `v1.pods {...}`
+You can browse auto-generated API documentation at the [project website](http://kubernetes.io/v1.0/api-ref.html) or on [github](../../docs/api-reference/).
```
Comment (caesarxuchao) [inline docs/user-guide/working-with-resources.md]: @bgrant0607 @erictune does this one needs to be cherry-picked?

**Comment by bgrant0607** [comment]
@caesarxuchao We could cherrypick it, but probably should change the link to v1.1. Do we have a way to automatically target the right version?

**Comment by caesarxuchao** [comment]
I figured out we need to do that change :)

No, it won't be automatically changed.

I can add it as a special case in versionized-docs.sh, though it will look ugly.

@nikhiljindal do you have better suggestions?

**Pull request merged**